lib/sax.js

```diff
@@ -554,6 +554,33 @@ function parseDCC(source,start,domBuilder,errorHandler){//sure start with '<!'
 			domBuilder.endCDATA() 
 			return end+3;
 		}
+		if(source.substr(start+2,6) == 'ENTITY'){
+			var end = source.indexOf('>', start+8);
+			var chunk = source.substring(start+8, end);
+			var match = chunk.match(/\s+(\w+)\s+(["'])/);
+			if (!match) {
+				errorHandler.error("Malformed internal entity declaration");
```

Comment:
Question: Would that trigger for all the "unsupported" DTD cases? 
Currently we are just ignoring those elements, if we now report an error, this might lead to "breaking behavior".

Which is a decision we can make, but we need to consider our options.

Or asked differently: Is there some other available entity syntax that we would consider "legit" but is actually not part of the supported cases and could cause any weird issues?

Reply:
Yes, you are absolutely right; there are other forms and only due to a sloppy regexp did this not trip up on those. I've corrected that in 0d665a564e69dabfd2eb89fe40cf802aa9c21bb4 and removed the comment in favour of just ignoring them for the time being.